File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/text/roles/TextColorRole.kt
package com.perrystreet.woof.designsystem.atomic.atoms.text.roles

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import com.perrystreet.woof.designsystem.theme.Theme

enum class TextColorRole {
    OnSurface {
        @Composable
        override fun color(): Color = Theme.colors.onSurface
    },
    OnSurfaceVariant {
        @Composable
        override fun color(): Color = Theme.colors.onSurfaceVariant
    },
    OnPrimary {
        @Composable
        override fun color(): Color = Theme.colors.onPrimary
    },
    OnScrim {
        @Composable
        override fun color(): Color = Theme.colors.onScrim
    },
    OnScrimVariant {
        @Composable
        override fun color(): Color = Theme.colors.onScrimVariant
    },
    Primary {
        @Composable
        override fun color(): Color = Theme.colors.primary
    },
    Destructive {
        @Composable
        override fun color(): Color = Theme.colors.destructive
    },
    ;

    @Composable
    abstract fun color(): Color
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Text/Roles/TextColorRole.swift
import SwiftUI

public enum TextColorRole {
    case onSurface
    case onSurfaceVariant
    case onPrimary
    case onScrim
    case onScrimVariant
    case primary
    case destructive

    public func color(from theme: ThemeImplementing) -> Color {
        switch self {
        case .onSurface: theme.colors.onSurface
        case .onSurfaceVariant: theme.colors.onSurfaceVariant
        case .onPrimary: theme.colors.onPrimary
        case .onScrim: theme.colors.onScrim
        case .onScrimVariant: theme.colors.onScrimVariant
        case .primary: theme.colors.primary
        case .destructive: theme.colors.destructive
        }
    }
}
